cohortextractor/query_engines/mssql.py

```diff
@@ -171,12 +170,21 @@ def get_query_nodes_from_category_definitions(self, definitions, query_nodes=Non
         """
         Get all the referenced query nodes for category definitions.  If a category
         definition (Comparator) has a LHS which is not itself a Comparator,
-        it will be a query node
+        it will be a query node or a function-generated node
         """
         query_nodes = query_nodes or set()
         for definition in definitions:
             if isinstance(definition.lhs, Value):
-                query_nodes.add(definition.lhs)
+                # A ValueFromFunction is not itself a QueryNode, so we look for referenced
+                # nodes in its arguments
+                parent_nodes = (
+                    definition.lhs.arguments
```

Comment:
Why do we never need to unwrap the arguments for a RHS?

Reply:
Because it's never a Value.  It's always either a simple str/int/date/None or a Comparator. If it's a Comparator, it gets fed back into this to get the parents from its LHS.

Reply:
I see. Thanks.